File: python/1291.py
from math import floor, log10
from typing import List


class Solution:
    def sequentialDigits(self, low: int, high: int) -> List[int]:
        results = []

        for num_digits in range(floor(log10(low)) + 1, floor(log10(high)) + 2):
            for start in range(10 - num_digits):
                integer = int("123456789"[start : start + num_digits])

                if integer > high:
                    return results
                elif integer < low:
                    continue
                else:
                    results.append(integer)

        return results

File: go/1291.go
package main

import (
	"math"
	"strconv"
)

func main() {
}

func sequentialDigits(low int, high int) []int {
	results := make([]int, 0)

	n_digits_low := int(math.Log10(float64(low))) + 1
	n_digits_high := int(math.Log10(float64(high))) + 1

	for n := n_digits_low; n <= n_digits_high; n++ {
		for start := 0; start < 10-n; start++ {
			integer, _ := strconv.Atoi("123456789"[start : start+n])

			if integer > high {
				return results
			} else if integer >= low {
				results = append(results, integer)
			}
		}
	}

	return results
}
